config/initializers/register_provider_plugins.rb

```diff
@@ -0,0 +1,3 @@
+Rails.application.railties.each do |railtie|
+  Vmdb::Plugins.instance.register_provider_plugin(railtie) if railtie.try(:manageiq_provider_plugin?)
```

Comment:
How about adding a convention like approach here too:

```ruby
if railtie.name.start_with? "ManageIQ::Providers::" || railtie.try(:manageiq_provider_plugin?)
```

this way all engines under this namespace get registered automatically